web_poet/rules.py

```diff
@@ -0,0 +1,350 @@
+        :param instead_of: The Page Object that should be `replaced`.
+        :param to_return: The Item Class holding the data returned by the Page Object.
+            This could be omitted as it could be derived from the ``Returns[ItemClass]``
+            or ``ItemPage[ItemClass]`` declaration of the Page Object. See
+            :ref:`item-classes` section. Code example in :ref:`combination` subsection.
+        :param exclude: The URLs over which the override should **not** happen.
+        :param priority: The resolution priority in case of `conflicting` rules.
+            A conflict happens when the ``include``, ``override``, and ``exclude``
+            parameters are the same. If so, the `highest priority` will be
+            chosen.
+        """
+
+        def wrapper(cls):
+
+            if overrides is not None:
+                msg = (
+                    "The 'overrides' parameter in @handle_urls is deprecated. "
+                    "Use the 'instead_of' parameter."
+                )
+                warnings.warn(msg, DeprecationWarning, stacklevel=2)
+
+            rule = ApplyRule(
+                for_patterns=Patterns(
+                    include=as_list(include),
+                    exclude=as_list(exclude),
+                    priority=priority,
+                ),
+                use=cls,
+                instead_of=instead_of or overrides,
+                to_return=to_return or get_item_cls(cls),
+                meta=kwargs,
+            )
+            # If it was already defined, we don't want to override it
+            if cls not in self:
+                self[cls] = rule
+            else:
+                warnings.warn(
+                    f"Multiple @handle_urls decorators for the same Page Object "
+                    f"are ignored in the same Registry. The following rule is "
+                    f"ignored:\n{rule}",
+                    stacklevel=2,
+                )
+
+            return cls
+
+        return wrapper
+
+    def get_rules(self) -> List[ApplyRule]:
+        """Returns all the :class:`~.ApplyRule` that were declared using
+        the ``@handle_urls`` decorator.
+
+        .. note::
+
+            Remember to consider calling :func:`~.web_poet.rules.consume_modules`
+            beforehand to recursively import all submodules which contains the
+            ``@handle_urls`` decorators from external Page Objects.
+        """
+        return list(self.values())
+
+    def get_overrides(self) -> List[ApplyRule]:
+        """Deprecated, use :meth:`~.PageObjectRegistry.get_rules` instead."""
+        msg = "The 'get_overrides' method is deprecated. Use 'get_rules' instead."
+        warnings.warn(msg, DeprecationWarning, stacklevel=2)
+        return self.get_rules()
+
+    def search_rules(self, **kwargs) -> List[ApplyRule]:
+        """Return any :class:`ApplyRule` that has any of its attributes
+        match the rules inside the registry.
```

Comment:
Could you please re-phrase it? I initially read it as "if any of the kwargs match attribute name of the rule, return the rule". But based on tests it seems all kwargs should be matched.

Reply:
Good catch! https://github.com/scrapinghub/web-poet/pull/84/commits/bce97be36bc4c6a6e2ca5889d3a2f2c2e440fef0